runner/app/live/trickle/encoder.py

```diff
@@ -82,35 +96,92 @@ def custom_io_open(url: str, flags: int, options: dict):
             frame.pts = rescale_ts(avframe.timestamp, avframe.time_base, output_video_stream.codec_context.time_base)
             frame.time_base = output_video_stream.codec_context.time_base
             current = avframe.timestamp * avframe.time_base
+            if len(audio_buffer) > 0:
+                # check if video is too far behind audio. if so, drop to try and catch up
+                audio_ts = audio_buffer[0].timestamp * audio_buffer[0].time_base
+                if audio_ts - current > AUDIO_BUFFER_SECS:
```

Comment:
Is it possible for the video to come before the audio as well? What would happen on that case?

Reply:
Yep, that case is handled just before writing the audio - https://github.com/livepeer/ai-runner/pull/572/files#diff-58ccb9520686dccb238caf0d92a8d842e08674435226cf5aeaaaf023bd17573fR169-R178

TBH a better implementation would queue up _either_ video or audio if we were expecting both, but I don't think we really want the additional latency on the video side here.